byzcoin/service.go

```diff
@@ -686,12 +682,22 @@ func (s *Service) createNewBlock(scID skipchain.SkipBlockID, r *onet.Roster, tx
 		return nil, errors.New("Couldn't marshal data: " + err.Error())
 	}
 
+	if r != nil {
+		sb.Roster = r
+	}
 	var ssb = skipchain.StoreSkipBlock{
 		NewBlock:          sb,
 		TargetSkipChainID: scID,
 	}
 	log.Lvlf3("Storing skipblock with %d transactions.", len(txRes))
-	ssbReply, err := s.skService().StoreSkipBlock(&ssb)
+	var ssbReply *skipchain.StoreSkipBlockReply
+	if sb.Roster.List[0].Equal(s.ServerIdentity()) {
```

Comment:
According to line 685, the roster could be nil. Either we check the function parameter to force it won't or this needs to be checked against nil pointer access.

Reply:
Added test in genesis-block creation.